core/control-plane/control-plane-api/src/main/java/org/eclipse/dataspaceconnector/controlplane/api/ControlPlaneApiExtension.java

```diff
@@ -0,0 +1,125 @@
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.controlplane.api;
+
+import org.eclipse.dataspaceconnector.api.auth.AuthenticationRequestFilter;
+import org.eclipse.dataspaceconnector.api.auth.AuthenticationService;
+import org.eclipse.dataspaceconnector.controlplane.api.transferprocess.TransferProcessControlApiController;
+import org.eclipse.dataspaceconnector.controlplane.api.transferprocess.model.TransferProcessFailStateDto;
+import org.eclipse.dataspaceconnector.runtime.metamodel.annotation.Extension;
+import org.eclipse.dataspaceconnector.runtime.metamodel.annotation.Inject;
+import org.eclipse.dataspaceconnector.runtime.metamodel.annotation.Provider;
+import org.eclipse.dataspaceconnector.runtime.metamodel.annotation.Provides;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.WebServer;
+import org.eclipse.dataspaceconnector.spi.WebService;
+import org.eclipse.dataspaceconnector.spi.system.Hostname;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+import org.eclipse.dataspaceconnector.spi.transfer.TransferProcessManager;
+import org.eclipse.dataspaceconnector.spi.transfer.callback.ControlPlaneApiUrl;
+import org.jetbrains.annotations.NotNull;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import static java.lang.String.format;
+
+/**
+ * {@link ControlPlaneApiExtension } exposes HTTP endpoints for internal interaction with the Control Plane
+ */
+@Provides(ControlPlaneApiUrl.class)
+@Extension(value = ControlPlaneApiExtension.NAME)
+public class ControlPlaneApiExtension implements ServiceExtension {
+
+    public static final String NAME = "Control Plane API";
+
+    public static final String CONTROL_PLANE_API_CONFIG = "web.http.controlplane";
+    public static final String CONTROL_PLANE_CONTEXT_ALIAS = "controlplane";
+
+    public static final int DEFAULT_CONTROL_PLANE_API_PORT = 8384;
+    public static final String DEFAULT_CONTROL_PLANE_API_CONTEXT_PATH = "/api/v1/controlplane";
+
+    private int port = DEFAULT_CONTROL_PLANE_API_PORT;
+    private String path = DEFAULT_CONTROL_PLANE_API_CONTEXT_PATH;
+
+    @Inject
+    private WebServer webServer;
+    @Inject
+    private WebService webService;
+    @Inject
+    private AuthenticationService authService;
+    @Inject
+    private TransferProcessManager transferProcessManager;
+    @Inject
+    private Hostname hostname;
+
+    @Override
+    public String name() {
+        return NAME;
+    }
+
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        var monitor = context.getMonitor();
+        var alias = CONTROL_PLANE_CONTEXT_ALIAS;
+
+        var config = context.getConfig(CONTROL_PLANE_API_CONFIG);
```

Comment:
This piece of code (getting the context config, adding port mapping and printing log) is used also in other classes as `IdsApiConfigurationExtension`, `DataManagementApiConfigurationExtension`, `HttpWebhookExtension`... could it be a good moment to extract an utility method? Otherwise a dedicated issue could be just enough for now.

Reply:
I would raise a refactoring issue, for this, otherwise i fear it would overcomplicare this PR

Reply:
Raise here https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/2128 :)